qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,434 @@
+            physical_qubits: A list of physical qubits for the experiment.
+            lengths: A list of RB sequences lengths.
+            sampling_algorithm: The sampling algorithm to use for generating
+                circuit layers. Defaults to "edge_grab" which uses :class:`.EdgeGrabSampler`.
+            start_end_clifford: If True, begin the circuit with uniformly random 1-qubit
+                Cliffords and end the circuit with their inverses.
+            pauli_randomize: If True, surround each sampled circuit layer with layers of
+                uniformly random 1-qubit Paulis.
+            two_qubit_gate_density: Expected proportion of qubit sites with two-qubit
+                gates over all circuit layers (not counting optional layers at the start
+                and end). Only has effect if the default sampler
+                :class:`.EdgeGrabSampler` is used.
+            two_qubit_gate: The two-qubit gate to use. Defaults to
+                :class:`~qiskit.circuit.library.CXGate`. Only has effect if the
+                default sampler :class:`.EdgeGrabSampler` is used.
+            num_samples: Number of samples to generate for each sequence length.
+            sampler_opts: Optional dictionary of keyword arguments to pass to the sampler.
+            backend: Optional, the backend to run the experiment on.
+            seed: Optional, seed used to initialize ``numpy.random.default_rng``.
+                when generating circuits. The ``default_rng`` will be initialized
+                with this seed value every time :meth:`circuits` is called.
+            full_sampling: If True all Cliffords are independently sampled for
+                all lengths. If False for sample of lengths longer sequences are
+                constructed by appending additional Clifford samples to shorter
+                sequences.
+            inverting_pauli_layer: If True, a layer of Pauli gates is appended at the
+                end of the circuit to set all qubits to 0.
+
+        Raises:
+            QiskitError: if an odd length or a negative two qubit gate density is provided
+        """
+
+        if not all(length % 2 == 0 for length in lengths):
+            raise QiskitError("All lengths must be even")
+
+        super().__init__(
+            physical_qubits,
+            lengths,
+            backend=backend,
+            num_samples=num_samples,
+            seed=seed,
+            full_sampling=full_sampling,
+        )
+
+        self.set_experiment_options(
+            sampling_algorithm=sampling_algorithm,
+            sampler_opts=sampler_opts,
+            start_end_clifford=start_end_clifford,
+            pauli_randomize=pauli_randomize,
+            two_qubit_gate=two_qubit_gate,
+            two_qubit_gate_density=two_qubit_gate_density,
+            inverting_pauli_layer=inverting_pauli_layer,
+        )
+
+        self._distribution = self.sampler_map.get(sampling_algorithm)(seed=seed, **sampler_opts)
```

Comment:
You need to catch key error with friendly message. `get` without default value is just directly accessing the attribute.